pkg/util/leases.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+	coordinationv1api "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+var timeNow = time.Now
+
+func IsLeaseValid(lease coordinationv1api.Lease) bool {
+	var lastRenewTime time.Time
+	if lease.Spec.RenewTime != nil {
+		lastRenewTime = lease.Spec.RenewTime.Time
+	} else if lease.Spec.AcquireTime != nil {
+		lastRenewTime = lease.Spec.AcquireTime.Time
+	} else {
+		klog.Warningf("lease %v has neither a renew time or an acquire time, marking as expired: %v", lease.Name, lease)
+		return false
+	}
+
+	duration := time.Duration(*lease.Spec.LeaseDurationSeconds) * time.Second
+
+	return lastRenewTime.Add(duration).After(timeNow()) // renewTime+duration > time.Now()
+}
+func NewLeaseFromTemplate(template LeaseTemplate) *coordinationv1api.Lease {
```

Comment:
Looks like this is only used for testing? If so, move it to the `testing` package: https://github.com/kubernetes-sigs/apiserver-network-proxy/tree/master/pkg/testing